The dataset is provided by: https://data.europa.eu/euodp/en/data/dataset/covid-19-coronavirus-data
Their intro to the dataset:

The dataset contains the latest available public data on COVID-19 including a daily situation update, the epidemiological curve and the global geographical distribution (EU/EEA and the UK, worldwide). On 12 February 2020, the novel coronavirus was named severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) while the disease associated with it is now referred to as COVID-19. ECDC is closely monitoring this outbreak and providing risk assessments to guide EU Member States and the EU Commission in their response activities.

# Import the necessary libraries:

In [1]:
import pandas as pd
import json
from pandas.io.json import json_normalize
import urllib.request, json 

# Load JSON data

In [2]:
jsonurl = urllib.request.urlopen("https://opendata.ecdc.europa.eu/covid19/casedistribution/json/")
text = json.loads(jsonurl.read()) 

text 

{'records': [{'dateRep': '05/04/2020',
   'day': '5',
   'month': '4',
   'year': '2020',
   'cases': '35',
   'deaths': '1',
   'countriesAndTerritories': 'Afghanistan',
   'geoId': 'AF',
   'countryterritoryCode': 'AFG',
   'popData2018': '37172386'},
  {'dateRep': '04/04/2020',
   'day': '4',
   'month': '4',
   'year': '2020',
   'cases': '0',
   'deaths': '0',
   'countriesAndTerritories': 'Afghanistan',
   'geoId': 'AF',
   'countryterritoryCode': 'AFG',
   'popData2018': '37172386'},
  {'dateRep': '03/04/2020',
   'day': '3',
   'month': '4',
   'year': '2020',
   'cases': '43',
   'deaths': '0',
   'countriesAndTerritories': 'Afghanistan',
   'geoId': 'AF',
   'countryterritoryCode': 'AFG',
   'popData2018': '37172386'},
  {'dateRep': '02/04/2020',
   'day': '2',
   'month': '4',
   'year': '2020',
   'cases': '26',
   'deaths': '0',
   'countriesAndTerritories': 'Afghanistan',
   'geoId': 'AF',
   'countryterritoryCode': 'AFG',
   'popData2018': '37172386'},
  {'dateRep': '01/

In [3]:
df = json_normalize(text, 'records')
df

cases countriesAndTerritories countryterritoryCode     dateRep day  \
0       35             Afghanistan                  AFG  05/04/2020   5   
1        0             Afghanistan                  AFG  04/04/2020   4   
2       43             Afghanistan                  AFG  03/04/2020   3   
3       26             Afghanistan                  AFG  02/04/2020   2   
4       25             Afghanistan                  AFG  01/04/2020   1   
5       27             Afghanistan                  AFG  31/03/2020  31   
6        8             Afghanistan                  AFG  30/03/2020  30   
7       15             Afghanistan                  AFG  29/03/2020  29   
8       16             Afghanistan                  AFG  28/03/2020  28   
9        0             Afghanistan                  AFG  27/03/2020  27   
10      33             Afghanistan                  AFG  26/03/2020  26   
11       2             Afghanistan                  AFG  25/03/2020  25   
12       6             Afghanistan                  AFG  24/03/2020  24   
13      10             Afghanistan                  AFG  23/03/2020  23   
14       0             Afghanistan                  AFG  22/03/2020  22   
15       2             Afghanistan                  AFG  21/03/2020  21   
16       0             Afghanistan                  AFG  20/03/2020  20   
17       0             Afghanistan                  AFG  19/03/2020  19   
18       1             Afghanistan                  AFG  18/03/2020  18   
19       5             Afghanistan                  AFG  17/03/2020  17   
20       6             Afghanistan                  AFG  16/03/2020  16   
21       3             Afghanistan                  AFG  15/03/2020  15   
22       3             Afghanistan                  AFG  11/03/2020  11   
23       3             Afghanistan                  AFG  08/03/2020   8   
24       0             Afghanistan                  AFG  02/03/2020   2   
25       0             Afghanistan                  AFG  01/03/2020   1   
26       0             Afghanistan                  AFG  29/02/2020  29   
27       0             Afghanistan                  AFG  28/02/2020  28   
28       0             Afghanistan                  AFG  27/02/2020  27   
29       0             Afghanistan                  AFG  26/02/2020  26   
...    ...                     ...                  ...         ...  ..   
8875     0                  Zambia                  ZMB  01/04/2020   1   
8876     6                  Zambia                  ZMB  31/03/2020  31   
8877     1                  Zambia                  ZMB  30/03/2020  30   
8878    12                  Zambia                  ZMB  29/03/2020  29   
8879     2                  Zambia                  ZMB  28/03/2020  28   
8880     2                  Zambia                  ZMB  27/03/2020  27   
8881     9                  Zambia                  ZMB  26/03/2020  26   
8882     0                  Zambia                  ZMB  25/03/2020  25   
8883     0                  Zambia                  ZMB  24/03/2020  24   
8884     1                  Zambia                  ZMB  23/03/2020  23   
8885     0                  Zambia                  ZMB  22/03/2020  22   
8886     0                  Zambia                  ZMB  21/03/2020  21   
8887     0                  Zambia                  ZMB  20/03/2020  20   
8888     2                  Zambia                  ZMB  19/03/2020  19   
8889     0                Zimbabwe                  ZWE  05/04/2020   5   
8890     1                Zimbabwe                  ZWE  04/04/2020   4   
8891     0                Zimbabwe                  ZWE  03/04/2020   3   
8892     0                Zimbabwe                  ZWE  02/04/2020   2   
8893     1                Zimbabwe                  ZWE  01/04/2020   1   
8894     0                Zimbabwe                  ZWE  31/03/2020  31   
8895     0                Zimbabwe                  ZWE  30/03/2020  30   
8896     2                Zimb

In [4]:
df.countriesAndTerritories.unique()

array(['Afghanistan', 'Albania', 'Andorra', 'Algeria', 'Angola',
       'Anguilla', 'Antigua_and_Barbuda', 'Argentina', 'Armenia', 'Aruba',
       'Australia', 'Austria', 'Azerbaijan', 'Bahamas', 'Bahrain',
       'Bangladesh', 'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin',
       'Bermuda', 'Bhutan', 'Bolivia',
       'Bonaire, Saint Eustatius and Saba', 'Bosnia_and_Herzegovina',
       'Botswana', 'Brazil', 'British_Virgin_Islands',
       'Brunei_Darussalam', 'Bulgaria', 'Burkina_Faso', 'Burundi',
       'Cambodia', 'Cameroon', 'Canada', 'Cape_Verde',
       'Cases_on_an_international_conveyance_Japan', 'Cayman_Islands',
       'Central_African_Republic', 'Chad', 'Chile', 'China', 'Colombia',
       'Congo', 'Costa_Rica', 'Cote_dIvoire', 'Croatia', 'Cuba',
       'Curaçao', 'Cyprus', 'Czech_Republic',
       'Democratic_Republic_of_the_Congo', 'Denmark', 'Djibouti',
       'Dominica', 'Dominican_Republic', 'Ecuador', 'Egypt',
       'El_Salvador', 'Equatorial_Guinea', 'Eritrea

I wanted to make sure United States was in there. (nationalistic ;p)

In [5]:
df.countriesAndTerritories.value_counts().head(10)

Netherlands       97
Sweden            97
Italy             97
United_Kingdom    97
Australia         97
Czech_Republic    97
Austria           97
Belgium           97
Canada            97
Spain             97
Name: countriesAndTerritories, dtype: int64

These are the countries with the most records/cases, which is 97 total.

Let's take a look at the timeframe of the date. We know that April 1st is included, but how far back does it go?

In [6]:
df.loc[df['dateRep'] == '01/01/2020'].head(10)

cases countriesAndTerritories countryterritoryCode     dateRep day  \
85       0             Afghanistan                  AFG  01/01/2020   1   
221      0                 Algeria                  DZA  01/01/2020   1   
388      0                 Armenia                  ARM  01/01/2020   1   
499      0               Australia                  AUS  01/01/2020   1   
596      0                 Austria                  AUT  01/01/2020   1   
686      0              Azerbaijan                  AZE  01/01/2020   1   
801      0                 Bahrain                  BHR  01/01/2020   1   
931      0                 Belarus                  BLR  01/01/2020   1   
1028     0                 Belgium                  BEL  01/01/2020   1   
1252     0                  Brazil                  BRA  01/01/2020   1   

     deaths geoId month popData2018  year  
85        0    AF     1    37172386  2020  
221       0    DZ     1    42228429  2020  
388       0    AM     1     2951776  2020  
499       0    AU     1    24992369  2020  
596       0    AT     1     8847037  2020  
686       0    AZ     1     9942334  2020  
801       0    BH     1     1569439  2020  
931       0    BY     1     9485386  2020  
1028      0    BE     1    11422068  2020  
1252      0    BR     1   209469333  2020

In [22]:
df.loc[df['dateRep'] == '05/04/2020'].head(10)

cases countriesAndTerritories countryterritoryCode     dateRep day deaths  \
0      35             Afghanistan                  AFG  05/04/2020   5      1   
87     29                 Albania                  ALB  05/04/2020   5      2   
115    27                 Andorra                  AND  05/04/2020   5      1   
131   314                 Algeria                  DZA  05/04/2020   5     47   
230     2                  Angola                  AGO  05/04/2020   5      0   
245     0                Anguilla                       05/04/2020   5      0   
255     0     Antigua_and_Barbuda                  ATG  05/04/2020   5      0   
272   186               Argentina                  ARG  05/04/2020   5      6   
302    34                 Armenia                  ARM  05/04/2020   5      0   
390     2                   Aruba                  ABW  05/04/2020   5      0   

    geoId month popData2018  year  
0      AF     4    37172386  2020  
87     AL     4     2866376  2020  
115    AD     4       77006  2020  
131    DZ     4    42228429  2020  
230    AO     4    30809762  2020  
245    AI     4              2020  
255    AG     4       96286  2020  
272    AR     4    44494502  2020  
302    AM     4     2951776  2020  
390    AW     4      105845  2020

(Found endpoints in range this way because dates are in European format with days first. And .min() and .max() reads from left to right.) So date ranges from Jan 1st, 2020 to today, April 5th.

Now, let us take a look at which countries had the most cases and which countries had the most deaths.

# EDA

Let us import my fav viz library --> plotly! along with some other helpful libraries. =)

In [8]:
import numpy as np 

import plotly.offline as py
py.init_notebook_mode(connected=True)
import plotly.graph_objs as go
import plotly.tools as tls
import plotly.express as px

pd.options.mode.chained_assignment = None
pd.options.display.max_columns = 999

In [9]:
#This is just for REFERENCE. What are the columns?
list(df)

['cases',
 'countriesAndTerritories',
 'countryterritoryCode',
 'dateRep',
 'day',
 'deaths',
 'geoId',
 'month',
 'popData2018',
 'year']

## Each country's relative population

In [10]:
df2 = df.drop_duplicates(subset='countriesAndTerritories', keep="last")

In [11]:
df2.head()

cases countriesAndTerritories countryterritoryCode     dateRep day deaths  \
86      0             Afghanistan                  AFG  31/12/2019  31      0   
114     2                 Albania                  ALB  09/03/2020   9      0   
222     0                 Algeria                  DZA  31/12/2019  31      0   
229     1                 Andorra                  AND  03/03/2020   3      0   
244     2                  Angola                  AGO  22/03/2020  22      0   

    geoId month popData2018  year  
86     AF    12    37172386  2019  
114    AL     3     2866376  2020  
222    DZ    12    42228429  2019  
229    AD     3       77006  2020  
244    AO     3    30809762  2020

In [12]:
x = df2['countriesAndTerritories']
y = df2['popData2018']


fig = go.Figure(data=[go.Bar(
            x=x, y=y,
            text=y,
            textposition='auto'
        )])

fig.show()

If you hover over the top 3 tallest bars, as expected, China, India and United States are the most populous countries worldwide. This is something to keep note of as case numbers continue to increase.

# Get animations for #Cases and #Deaths over time (Across each day)

# Need to change dates from European to US...

In [13]:
# fig = px.line(df, x='dateRep', y='cases')
# fig.show()

# Meanwhile, in NY:

In [14]:
null = None
NY_now = {'latest' :[{"date":20200402,"state":"NY","positive":92381,"negative":146584,"pending":null,"hospitalizedCurrently":13383,"hospitalizedCumulative":20817,"inIcuCurrently":3396,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":7434,"hash":"764d0566c27be04c416c502640d5fffbcb8cad26","dateChecked":"2020-04-02T20:00:00Z","death":2373,"hospitalized":20817,"total":238965,"totalTestResults":238965,"posNeg":238965,"fips":"36","deathIncrease":432,"hospitalizedIncrease":2449,"negativeIncrease":9416,"positiveIncrease":8669,"totalTestResultsIncrease":18085},{"date":20200401,"state":"NY","positive":83712,"negative":137168,"pending":null,"hospitalizedCurrently":12226,"hospitalizedCumulative":18368,"inIcuCurrently":3022,"inIcuCumulative":3022,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":6142,"hash":"bc022c8f01dbc1a4d1f046ebce21c18a7b42cfa6","dateChecked":"2020-04-01T20:00:00Z","death":1941,"hospitalized":18368,"total":220880,"totalTestResults":220880,"posNeg":220880,"fips":"36","deathIncrease":391,"hospitalizedIncrease":2464,"negativeIncrease":7777,"positiveIncrease":7917,"totalTestResultsIncrease":15694},{"date":20200331,"state":"NY","positive":75795,"negative":129391,"pending":null,"hospitalizedCurrently":10929,"hospitalizedCumulative":15904,"inIcuCurrently":2710,"inIcuCumulative":2710,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":4975,"hash":"a3afdea93fae4daf8073978749d736d46e9134c6","dateChecked":"2020-03-31T20:00:00Z","death":1550,"hospitalized":15904,"total":205186,"totalTestResults":205186,"posNeg":205186,"fips":"36","deathIncrease":332,"hospitalizedIncrease":2183,"negativeIncrease":9420,"positiveIncrease":9298,"totalTestResultsIncrease":18718},{"date":20200330,"state":"NY","positive":66497,"negative":119971,"pending":null,"hospitalizedCurrently":9517,"hospitalizedCumulative":13721,"inIcuCurrently":2352,"inIcuCumulative":2352,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":4204,"hash":"a749060bc47c2573b505c13198673b348ed3cc2b","dateChecked":"2020-03-30T20:00:00Z","death":1218,"hospitalized":13721,"total":186468,"totalTestResults":186468,"posNeg":186468,"fips":"36","deathIncrease":253,"hospitalizedIncrease":1646,"negativeIncrease":7124,"positiveIncrease":6984,"totalTestResultsIncrease":14108},{"date":20200329,"state":"NY","positive":59513,"negative":112847,"pending":null,"hospitalizedCurrently":8503,"hospitalizedCumulative":12075,"inIcuCurrently":2037,"inIcuCumulative":2037,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":3572,"hash":"b5656122e28105ae4788a0d68eabb1ef68a06287","dateChecked":"2020-03-29T20:00:00Z","death":965,"hospitalized":12075,"total":172360,"totalTestResults":172360,"posNeg":172360,"fips":"36","deathIncrease":237,"hospitalizedIncrease":2021,"negativeIncrease":9231,"positiveIncrease":7195,"totalTestResultsIncrease":16426},{"date":20200328,"state":"NY","positive":52318,"negative":103616,"pending":null,"hospitalizedCurrently":7328,"hospitalizedCumulative":10054,"inIcuCurrently":1755,"inIcuCumulative":1755,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":2726,"hash":"7592db95575a5d4ef226ff6014283d8d85a0db52","dateChecked":"2020-03-28T20:00:00Z","death":728,"hospitalized":10054,"total":155934,"totalTestResults":155934,"posNeg":155934,"fips":"36","deathIncrease":209,"hospitalizedIncrease":1528,"negativeIncrease":2498,"positiveIncrease":7683,"totalTestResultsIncrease":10181},{"date":20200327,"state":"NY","positive":44635,"negative":101118,"pending":null,"hospitalizedCurrently":6481,"hospitalizedCumulative":8526,"inIcuCurrently":1583,"inIcuCumulative":1583,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":2045,"hash":"d7f2dd3fcfbabebfdaeaa0a19e5db38d94fb112a","dateChecked":"2020-03-27T20:00:00Z","death":519,"hospitalized":8526,"total":145753,"totalTestResults":145753,"posNeg":145753,"fips":"36","deathIncrease":134,"hospitalizedIncrease":1682,"negativeIncrease":16272,"positiveIncrease":7377,"totalTestResultsIncrease":23649},{"date":20200326,"state":"NY","positive":37258,"negative":84846,"pending":null,"hospitalizedCurrently":5327,"hospitalizedCumulative":6844,"inIcuCurrently":1290,"inIcuCumulative":1290,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"c9e5180c39aab68a5662116651d17a05f4e4b966","dateChecked":"2020-03-26T20:00:00Z","death":385,"hospitalized":6844,"total":122104,"totalTestResults":122104,"posNeg":122104,"fips":"36","deathIncrease":100,"hospitalizedIncrease":3039,"negativeIncrease":12178,"positiveIncrease":6447,"totalTestResultsIncrease":18625},{"date":20200325,"state":"NY","positive":30811,"negative":72668,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":3805,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"332b18596170cd3e35617cfba48a5723aa2ec8f3","dateChecked":"2020-03-25T20:00:00Z","death":285,"hospitalized":3805,"total":103479,"totalTestResults":103479,"posNeg":103479,"fips":"36","deathIncrease":75,"hospitalizedIncrease":571,"negativeIncrease":7063,"positiveIncrease":5146,"totalTestResultsIncrease":12209},{"date":20200324,"state":"NY","positive":25665,"negative":65605,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":3234,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"ea3323c8187b8ff5574d3576f83791e01dd1521f","dateChecked":"2020-03-24T20:00:00Z","death":210,"hospitalized":3234,"total":91270,"totalTestResults":91270,"posNeg":91270,"fips":"36","deathIncrease":96,"hospitalizedIncrease":599,"negativeIncrease":8191,"positiveIncrease":4790,"totalTestResultsIncrease":12981},{"date":20200323,"state":"NY","positive":20875,"negative":57414,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":2635,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"40440cdc2bc6ecd1d88040f01ebbdcb130a33257","dateChecked":"2020-03-23T20:00:00Z","death":114,"hospitalized":2635,"total":78289,"totalTestResults":78289,"posNeg":78289,"fips":"36","deathIncrease":0,"hospitalizedIncrease":661,"negativeIncrease":11181,"positiveIncrease":5707,"totalTestResultsIncrease":16888},{"date":20200322,"state":"NY","positive":15168,"negative":46233,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":1974,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"0df3f76e6407f511b0c5bf4f5647cce8f407250e","dateChecked":"2020-03-22T20:00:00Z","death":114,"hospitalized":1974,"total":61401,"totalTestResults":61401,"posNeg":61401,"fips":"36","deathIncrease":70,"hospitalizedIncrease":371,"negativeIncrease":11152,"positiveIncrease":4812,"totalTestResultsIncrease":15964},{"date":20200321,"state":"NY","positive":10356,"negative":35081,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":1603,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"eb983c3fb4fc2d7f3c2c1250578379f502787c29","dateChecked":"2020-03-21T20:00:00Z","death":44,"hospitalized":1603,"total":45437,"totalTestResults":45437,"posNeg":45437,"fips":"36","deathIncrease":9,"hospitalizedIncrease":1603,"negativeIncrease":9756,"positiveIncrease":3254,"totalTestResultsIncrease":13010},{"date":20200320,"state":"NY","positive":7102,"negative":25325,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":null,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"5293b5c5920709d4f3cf66b901621a61a47d0d23","dateChecked":"2020-03-20T20:00:00Z","death":35,"hospitalized":null,"total":32427,"totalTestResults":32427,"posNeg":32427,"fips":"36","deathIncrease":23,"hospitalizedIncrease":0,"negativeIncrease":7193,"positiveIncrease":2950,"totalTestResultsIncrease":10143},{"date":20200319,"state":"NY","positive":4152,"negative":18132,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":null,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"2eabc9b73139066e3613021b1e9c9deaf8af733c","dateChecked":"2020-03-19T20:00:00Z","death":12,"hospitalized":null,"total":22284,"totalTestResults":22284,"posNeg":22284,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":5917,"positiveIncrease":1770,"totalTestResultsIncrease":7687},{"date":20200318,"state":"NY","positive":2382,"negative":12215,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":null,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"65a08e6a0f81bd2c4bbc9988a17f0892d2ec4d35","dateChecked":"2020-03-18T20:00:00Z","death":12,"hospitalized":null,"total":14597,"totalTestResults":14597,"posNeg":14597,"fips":"36","deathIncrease":5,"hospitalizedIncrease":0,"negativeIncrease":6709,"positiveIncrease":682,"totalTestResultsIncrease":7391},{"date":20200317,"state":"NY","positive":1700,"negative":5506,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":null,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"a2d96e7a962926fa3165aef038360c097b6be4ce","dateChecked":"2020-03-17T20:00:00Z","death":7,"hospitalized":null,"total":7206,"totalTestResults":7206,"posNeg":7206,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":963,"positiveIncrease":750,"totalTestResultsIncrease":1713},{"date":20200316,"state":"NY","positive":950,"negative":4543,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":null,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"994a65ba75be3681f76ff4612eb8fde85cff81cd","dateChecked":"2020-03-16T20:00:00Z","death":7,"hospitalized":null,"total":5493,"totalTestResults":5493,"posNeg":5493,"fips":"36","deathIncrease":4,"hospitalizedIncrease":0,"negativeIncrease":0,"positiveIncrease":221,"totalTestResultsIncrease":221},{"date":20200315,"state":"NY","positive":729,"negative":4543,"pending":null,"hospitalizedCurrently":null,"hospitalizedCumulative":null,"inIcuCurrently":null,"inIcuCumulative":null,"onVentilatorCurrently":null,"onVentilatorCumulative":null,"recovered":null,"hash":"963f632b5d1577cd5cf9b6c332a912c2fdebea16","dateChecked":"2020-03-15T20:00:00Z","death":3,"hospitalized":null,"total":5272,"totalTestResults":5272,"posNeg":5272,"fips":"36","deathIncrease":3,"hospitalizedIncrease":0,"negativeIncrease":1764,"positiveIncrease":205,"totalTestResultsIncrease":1969},{"date":20200314,"state":"NY","positive":524,"negative":2779,"hash":"f28c39ca671c26d022ff2556494ad8767765eb93","dateChecked":"2020-03-14T20:00:00Z","total":3303,"totalTestResults":3303,"posNeg":3303,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":0,"positiveIncrease":103,"totalTestResultsIncrease":103},{"date":20200313,"state":"NY","positive":421,"negative":2779,"hash":"a0a0e2997c411014ef2cd98b9502387730f585c7","dateChecked":"2020-03-13T20:00:00Z","total":3200,"totalTestResults":3200,"posNeg":3200,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":2687,"positiveIncrease":205,"totalTestResultsIncrease":2892},{"date":20200312,"state":"NY","positive":216,"negative":92,"hash":"78ea2217b81c0eba25ec04b1196199ad81793ead","dateChecked":"2020-03-12T20:00:00Z","total":308,"totalTestResults":308,"posNeg":308,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":0,"positiveIncrease":0,"totalTestResultsIncrease":0},{"date":20200311,"state":"NY","positive":216,"negative":92,"hash":"6fbded3854324c39522ab7fdb01d2087f60e2d2b","dateChecked":"2020-03-11T20:00:00Z","total":308,"totalTestResults":308,"posNeg":308,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":0,"positiveIncrease":43,"totalTestResultsIncrease":43},{"date":20200310,"state":"NY","positive":173,"negative":92,"hash":"193f85893cecb99d063a03060bc4802d2d6458f1","dateChecked":"2020-03-10T20:00:00Z","total":265,"totalTestResults":265,"posNeg":265,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":0,"positiveIncrease":31,"totalTestResultsIncrease":31},{"date":20200309,"state":"NY","positive":142,"negative":92,"hash":"260a070d28ed40a35e8b31aca5f4ab3352dc9ba8","dateChecked":"2020-03-09T20:00:00Z","total":234,"totalTestResults":234,"posNeg":234,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":0,"positiveIncrease":37,"totalTestResultsIncrease":37},{"date":20200308,"state":"NY","positive":105,"negative":92,"hash":"f6d3b78f41cadeaf08e72f35692a9619d70aad07","dateChecked":"2020-03-08T20:00:00Z","total":197,"totalTestResults":197,"posNeg":197,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":0,"positiveIncrease":29,"totalTestResultsIncrease":29},{"date":20200307,"state":"NY","positive":76,"negative":92,"pending":236,"hash":"53a82f90b4d4a28a5b06248b1e4c78e9229312dd","dateChecked":"2020-03-07T21:00:00Z","total":404,"totalTestResults":168,"posNeg":168,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":0,"positiveIncrease":43,"totalTestResultsIncrease":43},{"date":20200306,"state":"NY","positive":33,"negative":92,"pending":236,"hash":"18e7306085561597629e7ffac110572b1aa9f9e5","dateChecked":"2020-03-06T21:00:00Z","total":361,"totalTestResults":125,"posNeg":125,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":16,"positiveIncrease":11,"totalTestResultsIncrease":27},{"date":20200305,"state":"NY","positive":22,"negative":76,"pending":24,"hash":"a5eb1a3c895ee3acd8c87b19363bab6d581b3933","dateChecked":"2020-03-05T21:00:00Z","total":122,"totalTestResults":98,"posNeg":98,"fips":"36","deathIncrease":0,"hospitalizedIncrease":0,"negativeIncrease":28,"positiveIncrease":16,"totalTestResultsIncrease":44},{"date":20200304,"state":"NY","positive":6,"negative":48,"pending":24,"hash":"0e095e2b3db515b00d892276e6cac6912b1e0111","dateChecked":"2020-03-04T21:00:00Z","total":78,"totalTestResults":54,"posNeg":54,"fips":"36","deathIncrease":null,"hospitalizedIncrease":null,"negativeIncrease":null,"positiveIncrease":null,"totalTestResultsIncrease":null}]}

In [15]:
df3 = json_normalize(NY_now,'latest')
df3

date           dateChecked   death  deathIncrease fips  \
0   20200402  2020-04-02T20:00:00Z  2373.0          432.0   36   
1   20200401  2020-04-01T20:00:00Z  1941.0          391.0   36   
2   20200331  2020-03-31T20:00:00Z  1550.0          332.0   36   
3   20200330  2020-03-30T20:00:00Z  1218.0          253.0   36   
4   20200329  2020-03-29T20:00:00Z   965.0          237.0   36   
5   20200328  2020-03-28T20:00:00Z   728.0          209.0   36   
6   20200327  2020-03-27T20:00:00Z   519.0          134.0   36   
7   20200326  2020-03-26T20:00:00Z   385.0          100.0   36   
8   20200325  2020-03-25T20:00:00Z   285.0           75.0   36   
9   20200324  2020-03-24T20:00:00Z   210.0           96.0   36   
10  20200323  2020-03-23T20:00:00Z   114.0            0.0   36   
11  20200322  2020-03-22T20:00:00Z   114.0           70.0   36   
12  20200321  2020-03-21T20:00:00Z    44.0            9.0   36   
13  20200320  2020-03-20T20:00:00Z    35.0           23.0   36   
14  20200319  2020-03-19T20:00:00Z    12.0            0.0   36   
15  20200318  2020-03-18T20:00:00Z    12.0            5.0   36   
16  20200317  2020-03-17T20:00:00Z     7.0            0.0   36   
17  20200316  2020-03-16T20:00:00Z     7.0            4.0   36   
18  20200315  2020-03-15T20:00:00Z     3.0            3.0   36   
19  20200314  2020-03-14T20:00:00Z     NaN            0.0   36   
20  20200313  2020-03-13T20:00:00Z     NaN            0.0   36   
21  20200312  2020-03-12T20:00:00Z     NaN            0.0   36   
22  20200311  2020-03-11T20:00:00Z     NaN            0.0   36   
23  20200310  2020-03-10T20:00:00Z     NaN            0.0   36   
24  20200309  2020-03-09T20:00:00Z     NaN            0.0   36   
25  20200308  2020-03-08T20:00:00Z     NaN            0.0   36   
26  20200307  2020-03-07T21:00:00Z     NaN            0.0   36   
27  20200306  2020-03-06T21:00:00Z     NaN            0.0   36   
28  20200305  2020-03-05T21:00:00Z     NaN            0.0   36   
29  20200304  2020-03-04T21:00:00Z     NaN            NaN   36   

                                        hash  hospitalized  \
0   764d0566c27be04c416c502640d5fffbcb8cad26       20817.0   
1   bc022c8f01dbc1a4d1f046ebce21c18a7b42cfa6       18368.0   
2   a3afdea93fae4daf8073978749d736d46e9134c6       15904.0   
3   a749060bc47c2573b505c13198673b348ed3cc2b       13721.0   
4   b5656122e28105ae4788a0d68eabb1ef68a06287       12075.0   
5   7592db95575a5d4ef226ff6014283d8d85a0db52       10054.0   
6   d7f2dd3fcfbabebfdaeaa0a19e5db38d94fb112a        8526.0   
7   c9e5180c39aab68a5662116651d17a05f4e4b966        6844.0   
8   332b18596170cd3e35617cfba48a5723aa2ec8f3        3805.0   
9   ea3323c8187b8ff5574d3576f83791e01dd1521f        3234.0   
10  40440cdc2bc6ecd1d88040f01ebbdcb130a33257        2635.0   
11  0df3f76e6407f511b0c5bf4f5647cce8f407250e        1974.0   
12  eb983c3fb4fc2d7f3c2c1250578379f502787c29        1603.0   
13  5293b5c5920709d4f3cf66b901621a61a47d0d23           NaN   
14  2eabc9b73139066e3613021b1e9c9deaf8af733c           NaN   
15  65a08e6a0f81bd2c4bbc9988a17f0892d2ec4d35           NaN   
16  a2d96e7a962926fa3165aef038360c097b6be4ce           NaN   
17  994a65ba75be3681f76ff4612eb8fde85cff81cd           NaN   
18  963f632b5d1577cd5cf9b6c332a912c2fdebea16           NaN   
19  f28c39ca671c26d022ff2556494ad8767765eb93           NaN   
20  a0a0e2997c411014ef2cd98b9502387730f585c7           NaN   
21  78ea2217b81c0eba25ec04b1196199ad81793ead           NaN   
22  6fbded3854324c39522ab7fdb01d2087f60e2d2b           NaN   
23  193f85893cecb99d063a03060bc4802d2d6458f1           NaN   
24  260a070d28ed40a35e8b31aca5f4ab3352dc9ba8           NaN   
25  f6d3b78f41cadeaf08e72f35692a9619d70aad07           NaN   
26  53a82f90b4d4a28a5b06248b1e4c78e9229312dd           NaN   
27  18e7306085561597629e7ffac110572b1aa9f9e5           NaN   
28  a5eb1a3c895ee3acd8c87b19363bab6d581b3933           NaN   
29  0e095e2b3db515b00d892276e6cac6912b1e0111           NaN   

    hospitalizedCumulative  hospitaliz

Let's see how saturated NY hospitals still are:

In [16]:
fig = px.line(df3, x='dateChecked', y='hospitalizedCurrently')
fig.show()

It's NOT over. The curve has not flattened. There is no need to lift lockdown. More evidence below:

In [17]:
fig = px.line(df3, x='dateChecked', y='death')
fig.show()

This is the increase trend of deaths:

In [18]:
fig = px.line(df3, x='dateChecked', y='deathIncrease')
fig.show()

There were some dips between March 20 and 23 and Match 23 and March 26...so not a continuous upward tick.

Historical data:

In [19]:
historical = pd.read_csv("daily.csv")
historical

date state  positive  negative  pending  hospitalizedCurrently  \
0     20200402    AK     143.0    4879.0      NaN                    NaN   
1     20200402    AL    1233.0    7503.0      NaN                    NaN   
2     20200402    AR     643.0    7880.0      NaN                   66.0   
3     20200402    AS       0.0      20.0      6.0                    NaN   
4     20200402    AZ    1598.0   21111.0      NaN                    NaN   
5     20200402    CA    9191.0   23809.0  59500.0                 1922.0   
6     20200402    CO    3342.0   15303.0      NaN                    NaN   
7     20200402    CT    3824.0   14476.0      NaN                  827.0   
8     20200402    DC     653.0    4417.0      NaN                    NaN   
9     20200402    DE     393.0    4566.0      NaN                   56.0   
10    20200402    FL    8010.0   69286.0   1285.0                    NaN   
11    20200402    GA    5348.0   17609.0      NaN                    NaN   
12    20200402    GU      82.0     442.0      NaN                    NaN   
13    20200402    HI     258.0   10206.0     18.0                    NaN   
14    20200402    IA     614.0    8054.0      NaN                   74.0   
15    20200402    ID     669.0    6613.0      NaN                    NaN   
16    20200402    IL    7695.0   35961.0      NaN                    NaN   
17    20200402    IN    3039.0   13246.0      NaN                    NaN   
18    20200402    KS     552.0    6059.0      NaN                    NaN   
19    20200402    KY     680.0    7220.0      NaN                    NaN   
20    20200402    LA    9150.0   41936.0      NaN                    NaN   
21    20200402    MA    8966.0   47642.0      NaN                    NaN   
22    20200402    MD    2331.0   18890.0      NaN                    NaN   
23    20200402    ME     376.0    6088.0      NaN                    NaN   
24    20200402    MI   10791.0   11893.0      NaN                    NaN   
25    20200402    MN     742.0   21652.0      NaN                   75.0   
26    20200402    MO    1834.0   17849.0      NaN                    NaN   
27    20200402    MP       8.0      13.0     12.0                    NaN   
28    20200402    MS    1177.0    4753.0      NaN                    NaN   
29    20200402    MT     227.0    5093.0      NaN                    NaN   
...        ...   ...       ...       ...      ...                    ...   
1511  20200305    MD       0.0      17.0     14.0                    NaN   
1512  20200305    MI       0.0       8.0      0.0                    NaN   
1513  20200305    NC       1.0       NaN      NaN                    NaN   
1514  20200305    NE       0.0      17.0      5.0                    NaN   
1515  20200305    NH       2.0      16.0      4.0                    NaN   
1516  20200305    NJ       1.0       NaN      NaN                    NaN   
1517  20200305    NV       1.0      14.0      NaN                    NaN   
1518  20200305    NY      22.0      76.0     24.0                    NaN   
1519  20200305    OH       0.0       7.0      3.0                    NaN   
1520  20200305    OR       3.0      45.0     13.0                    NaN   
1521  20200305    SC       0.0       5.0      0.0                    NaN   
1522  20200305    TN       1.0       NaN      NaN                    NaN   
1523  20200305    TX       1.0       NaN      NaN                    NaN   
1524  20200305    VA       0.0      18.0      3.0                    NaN   
1525  20200305    WA      70.0       NaN      NaN                    NaN   
1526  20200305    WI       1.0      19.0      6.0                    NaN   
1527  20200304    AZ       2.0      27.0      5.0                    NaN   
1528  20200304    CA      53.0     462.0      NaN                    NaN   
1529  20200304    FL       2.0      24.0     16.0                    NaN   
1530  20200304    GA       2.0       NaN      NaN                    NaN   
1531  20200304    IL       4.0     124.0     27.0         

In [20]:
historical.shape

(1541, 25)

In [24]:
historical["onVentilatorCumulative"].value_counts().sort_index(ascending=False)

507.0    1
490.0    1
438.0    1
385.0    1
380.0    1
336.0    1
270.0    1
239.0    1
193.0    1
163.0    1
108.0    2
40.0     3
39.0     1
37.0     1
32.0     1
31.0     2
23.0     1
21.0     1
17.0     3
13.0     1
6.0      8
4.0      1
Name: onVentilatorCumulative, dtype: int64

At one point...there were 507 in serious condition on ventilators --> 507!